### PR TITLE
Add a create date to the playertimes table

### PR DESCRIPTION
**Pull request by alice-blue**
Add a create date column to the playertimes table, and automatically insert the date when a time is added.

(This PR includes only the two commits related to the create date.)

This addresses https://github.com/iftechfoundation/ifdb/issues/1155

### CODE REVIEW

File: sql/incoming-schema-changes.sql [hunk 1]
```diff
@@ -87,6 +87,7 @@ CREATE TABLE playertimes (
   userid VARCHAR(32) NOT NULL,
   time_in_minutes INT(5) unsigned not null,
   time_note VARCHAR(150),
+  createdate DATE,
```
Comment (salty-horse) [inline sql/incoming-schema-changes.sql:90]: I think `createdate DATE DEFAULT (CURRENT_DATE)` will make sure it's set automatically, so you even don't need to mention it when inserting a row.

(I haven't tested this)

**Comment by alice-blue** [comment]
I was wondering what to do about all of the existing rows that won't have a date. Do we need to do something like this?

```
UPDATE playertimes
SET createdate = '2014-11-23';
```

File: sql/incoming-schema-changes.sql (hunk 1 above)
Comment (alice-blue) [reply sql/incoming-schema-changes.sql:90]: [Stack Overflow says](https://stackoverflow.com/questions/20461030/current-date-curdate-not-working-as-default-date-value/51709343#51709343)

> According to this [documentation](https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html), starting in MySQL 8.0.13, you will be able to specify:
> 
> CREATE TABLE INVOICE(
>     INVOICEDATE DATE DEFAULT (CURRENT_DATE)
> )

I don't know what version we are using, but it sounds promising.

Comment (alice-blue) [reply sql/incoming-schema-changes.sql]: Ok, I changed this.

**Comment by dfabulich** [comment]
The new default value will take care of it automatically.

File: sql/incoming-schema-changes.sql [hunk 2]
```diff
@@ -87,6 +87,7 @@ CREATE TABLE playertimes (
   userid VARCHAR(32) NOT NULL,
   time_in_minutes INT(5) unsigned not null,
   time_note VARCHAR(150),
+  createdate DATE DEFAULT (CURRENT_DATE),
```
Comment (dfabulich) [inline sql/incoming-schema-changes.sql:90]: (Sorry, I thought I'd posted about this before, but I guess I didn't.)

```suggestion
  createdate DATETIME DEFAULT now(),
```

> https://stackoverflow.com/questions/31761047/what-difference-between-the-date-time-datetime-and-timestamp-types
> 
> **DATE:** It is used for values with a date part but no time part. MySQL retrieves and displays DATE values in ***YYYY-MM-DD*** format. The supported range is `1000-01-01` to `9999-12-31`.
> 
> **DATETIME:** It is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in ***YYYY-MM-DD HH:MM:SS*** format. The supported range is `1000-01-01 00:00:00` to `9999-12-31 23:59:59`.
> 
> **TIMESTAMP:** It is also used for values that contain both date and time parts, and includes the time zone. TIMESTAMP has a range of `1970-01-01 00:00:01` UTC to `2038-01-19 03:14:07` UTC.

**Review by dfabulich — CHANGES_REQUESTED** [review]

File: sql/incoming-schema-changes.sql (hunk 2 above)
Comment (alice-blue) [reply sql/incoming-schema-changes.sql:90]: What would we use the time for?

Comment (dfabulich) [reply sql/incoming-schema-changes.sql:90]: Sorting, at least. I would have used it for the deanonymization thingy.

Comment (alice-blue) [reply sql/incoming-schema-changes.sql]: Done

**Pull request merged**